config/opal_check_cuda.m4

```diff
@@ -50,9 +50,13 @@ AC_ARG_WITH([cuda],
             [AS_HELP_STRING([--with-cuda(=DIR)],
             [Build cuda support, optionally adding DIR/include])])
 AC_MSG_CHECKING([if --with-cuda is set])
+
+# Search for libcuda.so in $with_cuda if the user didn't pass --with-cuda-libdir
 AC_ARG_WITH([cuda-libdir],
             [AS_HELP_STRING([--with-cuda-libdir=DIR],
-                            [Search for CUDA libraries in DIR])])
+                            [Search for CUDA libraries in DIR])],
+            [],
+            [with_cuda_libdir=$(dirname $(find $with_cuda -name libcuda.so))])
```

Comment:
this needs to be improved to check for cases $with_cuda is not a directory

Reply:
It should work even if $with_cuda is empty. For example

```
[nsarkauskas@login01 ompi]$ echo $x

[nsarkauskas@login01 ompi]$ echo $with_cuda

[nsarkauskas@login01 ompi]$ x=$(dirname $(find $with_cuda -name libcuda.so))
dirname: missing operand
Try 'dirname --help' for more information.
[nsarkauskas@login01 ompi]$ echo $x

[nsarkauskas@login01 ompi]$
```